server/pbench/bin/pbench-base.sh

```diff
@@ -76,9 +76,17 @@ fi
 export mail_recipients=$(getconf.py mailto pbench-server)
 
 # make all the state directories for the pipeline and any others needed
-LINKDIRS="TODO TO-UNPACK TO-COPY-SOS TO-SYNC SYNCED TO-LINK TO-INDEX TO-BACKUP \
-	INDEXED WONT-INDEX DONE BAD-MD5 SATELLITE-MD5-PASSED SATELLITE-MD5-FAILED \
-	TO-DELETE SATELLITE-DONE TO-UNPACK UNPACKED MOVED-UNPACKED"
+# every related state directories are paired together with their
+# final state at the end
+LINKDIRS="TODO BAD-MD5 \
+    TO-UNPACK UNPACKED MOVED-UNPACKED  \
+    TO-SYNC SYNCED \
+    TO-LINK \
+    TO-INDEX INDEXED WONT-INDEX \
+    TO-COPY-SOS COPIED-SOS \
+    TO-BACKUP \
+    SATELLITE-MD5-PASSED SATELLITE-MD5-FAILED \
+    TO-DELETE SATELLITE-DONE"
```

Comment:
I performed the following to compare the two:
```
$ LINKDIRS="TODO TO-UNPACK TO-COPY-SOS TO-SYNC SYNCED TO-LINK TO-INDEX TO-BACKUP \
INDEXED WONT-INDEX DONE BAD-MD5 SATELLITE-MD5-PASSED SATELLITE-MD5-FAILED \
TO-DELETE SATELLITE-DONE TO-UNPACK UNPACKED MOVED-UNPACKED"
$ for i in $LINKDIRS ; do echo $i; done | sort > orig.linkdirs.lis
$ LINKDIRS="TODO BAD-MD5 \
     TO-UNPACK UNPACKED MOVED-UNPACKED  \
     TO-SYNC SYNCED \
     TO-LINK \
     TO-INDEX INDEXED WONT-INDEX \
     TO-COPY-SOS COPIED-SOS \
     TO-BACKUP \
     SATELLITE-MD5-PASSED SATELLITE-MD5-FAILED \
     TO-DELETE SATELLITE-DONE"
$ for i in $LINKDIRS ; do echo $i; done | sort > new.linkdirs.lis
$ diff -c orig.linkdirs.lis new.linkdirs.lis
*** orig.linkdirs.lis   2018-09-07 10:01:40.333072583 +0000
--- new.linkdirs.lis    2018-09-07 10:02:13.168546389 +0000
***************
*** 1,5 ****
  BAD-MD5
! DONE
  INDEXED
  MOVED-UNPACKED
  SATELLITE-DONE
--- 1,5 ----
  BAD-MD5
! COPIED-SOS
  INDEXED
  MOVED-UNPACKED
  SATELLITE-DONE
***************
*** 14,19 ****
  TO-LINK
  TO-SYNC
  TO-UNPACK
- TO-UNPACK
  UNPACKED
  WONT-INDEX
--- 14,18 ----
```

It looks like we added `COPIED-SOS`, but removed `DONE`, and removed the duplicate `TO-UNPACK`.

Do we still need the `DONE` state?

Reply:
@portante `DONE` is not being used by any of the server scripts, not anymore. That's why I removed it.

Reply:
@tenstormavi, great, can you be sure that is added to the commit comment?